### PR TITLE
remove the RemoveQuote API, and the remove quote p2p messages

### PR DESCRIPTION
**Pull request by cbeck88**
We think that the RemoveQuote API is not good to offer to clients, because ultimately, once they have signed a quote, nothing can guarantee that someone else won't match it, and in principle, someone can fork this code, and not remove quotes until they are actually expired. Then they could still be matched, even if we returned a "success" status on a RemoveQuote request.

In order to get users used to the idea that they have to cancel their quotes by sending a mobilecoin transaction, rather than just calling this API, we think we just shouldn't have this API.

Because we don't have this API, we also don't need to have this in the P2P gossip channels either -- all the P2P nodes will see that the quote expired because they are syncing the ledger, and that's their source of truth for whether it has truly expired.

We still send SciQuoteRemoved messages over the LiveUpdates channel. (Maybe we should call it SciQuoteExpired?)

This added some more tests that had to change after this change, and adjusted an e2e test that was testing that RemoveQuote was working.

We also added a new status code for `QUOTE_IS_STALE` to the SubmitQuote API, because otherwise stale quotes got back the non-specific `OTHER` code.

### CODE REVIEW

File: server/src/client_service.rs [hunk 1]
```diff
@@ -566,64 +580,69 @@ mod tests {
         let quotes = quote_book.get_quotes(&pair, .., 0).unwrap();
         assert_eq!(quotes.len(), 1);
 
-        // Remove it.
-        let quote_id = deqs_api::deqs::QuoteId::from(quotes[0].id());
-        let mut req = RemoveQuoteRequest::default();
-        req.set_quote_id(quote_id);
-
-        let resp = client_api.remove_quote(&req).expect("remove quote failed");
```
Comment (cbeck88) [inline server/src/client_service.rs:574]: the remove quote API was removed, so this goes away

File: server/src/client_service.rs [hunk 2]
```diff
@@ -566,64 +580,69 @@ mod tests {
...
-        let (client_api, _server, _msg_bus_rx) =
-            create_test_client_and_server(&quote_book, &logger);
 
-        let (tx, rx) = channel();
+        let mut ledger = create_and_initialize_test_ledger();
+        let starting_blocks = ledger.num_blocks().unwrap();
+
+        // Allow msg_bus_tx to be created later for remove_quote_callback,
+        // by putting Mutex<Option<Sender>> in the captured state of the callback
+        let remove_quote_callback_state = Arc::new(Mutex::new(Option::<Sender<Msg>>::default()));
```
Comment (cbeck88) [inline server/src/client_service.rs:603]: this is the trick i made up to work around the issue of server and quotebook both needing access to msg bus, without changing the other parts of the test infra,

it seems okay to me, but lmk if you think we should do this differently

Comment (eranrund) [reply server/src/client_service.rs:603]: Well I suppose if we are already simulating the delivery on the message bus, maybe you can skip the SynchronizedQuoteBook and just post directly to the message bus and see that it shows in the live update stream?

**Review by eranrund — APPROVED** [review]

**Comment by eranrund** [comment]
I'm okay with this going as is, but I think the test I commented on above can be simplified since we can just post to the message bus without dealing with a ledger.

File: server/src/client_service.rs (hunk 2 above)
Comment (cbeck88) [reply server/src/client_service.rs:603]: yeah i guess if this module is supposed to be testing the client api service, we can focus on just testing specifically that, and send the cancellation notice to `msg_bus_tx`. okay will fix

File: server/src/client_service.rs [hunk 3]
```diff
@@ -566,84 +580,135 @@ mod tests {
...
+            .collect::<Vec<_>>();
+        assert_eq!(received_scis, vec![]);
+
+        // If we try to add the same quote back, it should be rejected as stale
+        let req = SubmitQuotesRequest {
+            quotes: vec![(&sci).into()].into(),
+            ..Default::default()
+        };
+        let resp = client_api.submit_quotes(&req).expect("submit quote failed");
+        assert_eq!(resp.status_codes, vec![QuoteStatusCode::QUOTE_IS_STALE]);
```
Comment (cbeck88) [inline server/src/client_service.rs:700]: i added this test so that we can verify this, but it's no longer doing the funky remove_quote_callback

Comment (cbeck88) [reply server/src/client_service.rs:700]: I guess it's still possible to do this test without using the ledger, if we use like a mock synchronized ledger or something. but i feel like it's easier to just use a ledger and a real synchronized ledger

**Review by wjuan-mob — APPROVED** [review]

**Review by eranrund — APPROVED** [review]

**Pull request merged**